File: Projeto_2/src/seguroVeiculo.js
class SeguroVeiculo {
  constructor(cliente, carro, valor, vigencia) {
      this.cliente = cliente;
      this.carro = carro;
      this.valor = valor;
      this.vigencia = vigencia;
  }

  calcularValor(baseValor, taxa) {
      this.valor = baseValor + (baseValor * taxa);
      return this.valor;
  }

  verificarVigencia() {
      const hoje = new Date();
      const [inicio, fim] = this.vigencia.split(' a ');
      const vigenciaFinal = new Date(fim.split('/').reverse().join('-'));
      return hoje <= vigenciaFinal;
  }
}

class SeguroCarro extends SeguroVeiculo {
  calcularValor(baseValor, taxa) {
      taxa += 0.05;
      return super.calcularValor(baseValor, taxa);
  }
}

class SeguroMoto extends SeguroVeiculo {
  calcularValor(baseValor, taxa) {
      taxa += 0.03;
      return super.calcularValor(baseValor, taxa);
  }
}

export { SeguroVeiculo, SeguroCarro, SeguroMoto };

File: Projeto_2/src/seguro_veiculo.py
from datetime import date

class SeguroVeiculo:
    def __init__(self, cliente, valor, data_inicio, data_fim):
        self.cliente = cliente
        self.valor = valor
        self.data_inicio = data_inicio
        self.data_fim = data_fim

    def calcular_valor(self, base_valor, taxa):
        self.valor = base_valor * (1 + taxa)
        return self.valor

    def verificar_vigencia(self):
        hoje = date.today()
        return self.data_inicio <= hoje <= self.data_fim
